packages/gatsby/src/redux/types.ts

```diff
@@ -359,3 +362,18 @@ export interface ICreateRedirectAction {
   type: `CREATE_REDIRECT`
   payload: IRedirect
 }
+
+export interface IDeleteCacheAction {
+  type: `DELETE_CACHE`
+}
+
+export interface IUpdatePluginsHashAction {
+  type: `UPDATE_PLUGINS_HASH`
+  payload: IGatsbyState["status"]["PLUGINS_HASH"]
+}
+
+export interface ISetPluginStatusAction {
+  type: `SET_PLUGIN_STATUS`
+  plugin: IGatsbyPlugin
+  payload: Partial<IGatsbyPlugin>
```

Comment:
This is incorrect actually. Looking at the implementation, this is an arbitrary object of key value pairs. So it's typing sort of sucks.. but it's too deep to "fix". Let's make this 

```ts
payload: {
  [key: any]: any
}
```

Reply:
An index signature parameter type can only be either `string` or `number`, otherwise it is triggering an TS error.

Will implement `[key: string]: any` in this case 👌